cmd/swarm/main.go

```diff
@@ -181,6 +181,18 @@ var (
 		Usage:  "Number of recent chunks cached in memory (default 5000)",
 		EnvVar: SWARM_ENV_STORE_CACHE_CAPACITY,
 	}
+	SwarmResourceMultihashFlag = cli.BoolTFlag{
+		Name:  "multihash",
+		Usage: "Determines how to interpret data for a resource update. If not present, data will be interpreted as a multihash",
+	}
+	SwarmResourceNameFlag = cli.StringFlag{
+		Name:  "name",
+		Usage: "User-defined name for the new resource",
+	}
+	SwarmResourceDataOnCreateFlag = cli.StringFlag{
+		Name:  "data",
+		Usage: "Initializes the resource with the given hex-encoded data. Data must be prefixed by 0x",
```

Comment:
You can supply any kind of data, providing you hex-encode it first. In a command line, you don't want problems with character escaping and hex-encoding is very convenient. Another option would be to support base64 as well.

This is not meant to be final. The idea is that "serious" data comes in from stdin, but that'd be a new PR.

Reply:
Yes, we discussed the possibility of doing this `echo arbitrary.bin | swarm resource update`. I agree this would be very useful. However, I need to leave it for another PR, and would not be incompatble with having `--data "0x ..."` in the command line.